src/mix.cr

```diff
@@ -0,0 +1,255 @@
+      raise SDL::Error.new("Mix_QuerySpec: #{LibMIX.get_error}") if audio_open_count < 1
+      audio_open_count
+    end
+
+    def self.close
+      LibMIX.close_audio
+    end
+
+  end
+
+  module MIX
+    class Sample
+      protected getter sample
+
+      def self.decoder_count
+        LibMIX.get_num_chunk_decoders
+      end
+
+      def self.decoder_name(index)
+        LibMIX.get_chunk_decoder(index)
+      end
+
+      def initialize(filename = nil)
+        load filename if filename
+      end
+
+      def load(filename)
+        rwops = LibSDL.rw_from_file(filename, "rb")
+        @sample = LibMIX.load_wav_rw(rwops, 1)
+        raise SDL::Error.new("Mix_LoadWAV_RW") unless @sample
+      end
+
+      def finalize
+        LibMIX.free_chunk(sample)
+      end
+    end
+  end
+
+  module MIX
+    class Music
+      private getter music : Pointer(LibMIX::Music) | Nil
+
+      def initialize(filename = nil)
+        @music = filename ? load(filename) : nil
+      end
+
+      def play(repeats = -1)
+        LibMIX.play_music(music, repeats)
+      end
+
+      def pause
+        LibMIX.pause_music
+      end
+
+      def resume
+        LibMIX.resume_music
+      end
+
+      def stop
+        LibMIX.halt_music
+      end
+
+      def playing?
+        LibMIX.music_playing == 1
+      end
+
+      def paused?
+        LibMIX.music_paused == 1
+      end
+
+      def rewind
+        LibMIX.rewind_music
+      end
+
+      def fade_in(loops = -1, msec = 1000)
+        LibMIX.fade_in_music(music, loops, msec)
+      end
+
+      def fade_out(msec = 1000)
+        LibMIX.fade_out_music(msec)
+      end
+
+      def volume=(volume)
+        LibMIX.music_volume(volume > MAX_VOLUME ? MAX_VOLUME : volume)
+      end
+
+      def volume
+        LibMIX.music_volume(-1)
+      end
+
+      def load(filename, type : Type? = nil)
+        rwops = LibSDL.rw_from_file(filename, "rb")
+        @music = type ? load_music_type(rwops, type) : load_music(rwops)
+      end
+
+      def finalize
+        LibMIX.free_music(music)
+        @music = nil
+      end
+
+      private def load_music(rwops)
+        audio = LibMIX.load_mus_rw(rwops, 1)
+        raise SDL::Error.new("Mix_LoadMUSType_RW") unless audio
+        audio
+      end
+
+      private def load_music_type(rwops, typ)
+        audio = LibMIX.load_mus_type_rw(rwops, typ.to_s, 1)
+        raise SDL::Error.new("Mix_LoadMUS_RW") unless audio
+        audio
+      end
+    end
+  end
+
+  module MIX
+    class Channel
+      property id
+
+      def initialize(@id = 1)
+        @@count ||= 8 unless @@count
+      end
+
+      def self.allocate_channels(count)
+        @@count = count
+        LibMIX.allocate_channels(count)
+      end
+
+      def self.reserve_channels(count)
+        @@reserved = count
+        LibMIX.reserve_channels(count)
+      end
+
+      def self.play(smpl, repeats = 0)
+        LibMIX.play_channel(-1, smpl.sample, repeats)
```

Comment:
`MIX::Sample` shouldn't expose `#sample` but declare `#to_unsafe` so we can just pass an instance of `Sample` to C bindings, and Crystal will call `#to_unsafe` automatically. This way we could use `sample` as a variable name, which is nicer than `smpl`.